### ISSUE TITLE
Event Playlist Trigger Re-executing

### ISSUE THREAD

**Issue by GrizAF** [smeighan/xLights#1125]
EDIT:  As discussed on zoom, please ignore this first post and see the log for the crash when using the "jump to random step in specified playlist", plus the "send data when not play a sequence" check box. 
 I'll break into two reports next time.  Thanks again, Keith.  

Problem: Second and third methods of execution below are causing the lights to stay and never turn off due-to the event/playlist being re-exectued.  

Trying some more MIDI stuff, but I am not certain this is the cause (see LOR InputPup at bottom).  

The log file attached has three methods of triggering the event playlist "Mic" step "Tens" (first 10 pixels).

The first method at 2018-05-08 22:17:36,429 was executed by the button "play selected".  After to 3sec duration the lights turn off fine - works well. 

The second method at 2018-05-08 22:19:01,855 was executed with my microphone (via MIDI converter), and the lights remain on.  
2018-05-08 22:19:01,856 3224 log_base [INFO] ******** Playlist Mic [3.000] starting to play.
2018-05-08 22:19:01,856 3224 log_base [INFO]          ######## Playlist step Tens starting.
2018-05-08 22:19:01,856 3224 log_base [INFO]          ######## Playlist step Tens stopping.
2018-05-08 22:19:01,856 3224 log_base [INFO]          ######## Playlist step Tens starting.

I am not sure why the playlist step is starting again, but it does not check the sequence as does the first method.  Is this because of the "noisy" event being received?  The only way I can get the lights to turn off is to close xScheduler.  Turning off output to lights and Reset All Schedules does not turn off the lights.  

To be sure this isn't just a MIDI issue I disconnected the mic from my computer and ran the same event from my LOR InputPup and received a similar re-execution (third method; lights remain off and won't turn off):

2018-05-08 22:35:31,633 9536 log_base [DEBUG] Event fired LOR:IPP -> Run event playlist step:Mic,Tens
2018-05-08 22:35:31,634 9536 log_base [INFO] Playing event playlist Mic step Tens.
2018-05-08 22:35:31,634 9536 log_base [DEBUG] Copying playlist Mic [3.000] steps 1.
2018-05-08 22:35:31,634 9536 log_base [INFO] ******** Playlist Mic [3.000] starting to play.
2018-05-08 22:35:31,634 9536 log_base [INFO]          ######## Playlist step Tens starting.
2018-05-08 22:35:31,634 9536 log_base [INFO]          ######## Playlist step Tens stopping.
2018-05-08 22:35:31,634 9536 log_base [INFO]          ######## Playlist step Tens starting.
2018-05-08 22:35:31,634 9536 log_base [DEBUG]     Event processed.
2018-05-08 22:35:34,609 10468 log_base [INFO]          ######## Playlist step Tens stopping.
2018-05-08 22:35:40,838 10468 log_base [DEBUG] Viewing log file C:\Users\Home\AppData\Roaming/xSchedule_l4cpp.log. 

Appreciate any advice/thoughts on this one.  I'm also attaching the full log (tried to make it easier to read in the reexecute log.  Thanks for the help.

Josh

[Reexecute log.txt](https://github.com/smeighan/xLights/files/1986194/Reexecute.log.txt)
[Full Log.txt](https://github.com/smeighan/xLights/files/1986198/Full.Log.txt)

**Comment by GrizAF** [comment]
By the way - my processor has climbed back up to the high 30-40% mark.  Don't know if this is related to all of the events I have been trying out.  Sorry, but I really appreciate the help!

**Comment by GrizAF** [comment]
Along with the issues with the Run Event Playlist Step, I have also found consistent xscheduler crashes when using the "jump to random step in specified playlist".  I have more 6 - 7 steps in my sequence.  Log attached.  Thanks for the help.

[xSchedule_l4cpp.log](https://github.com/smeighan/xLights/files/1993813/xSchedule_l4cpp.log)

**Comment by GrizAF** [comment]
I wanted to follow up on this and see if you needed any more information from me on this.  Thanks again for all you do.

**Comment by keithsw1111** [comment]
Is this working now?

**Comment by GrizAF** [comment]
This is not crashing anymore, but the "jump to random step" function is still not working.  

I also have a few other bugs I have found in xscheduler events, but I wanted to discuss with you before writing them up.  These issues, including "jump to random step" may be a result of how I have structured my event playlists, but I have tried various methods/setups with the same results.  Any info is appreciated.
